presto-main/src/test/java/com/facebook/presto/operator/scalar/TestTDigestFunctions.java

```diff
@@ -0,0 +1,610 @@
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.tdigest.TDigest.createTDigest;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
+import static java.util.Collections.sort;
+
+public class TestTDigestFunctions
+        extends AbstractTestFunctions
+{
+    private static final int NUMBER_OF_ENTRIES = 1_000_000;
+    private static final int STANDARD_COMPRESSION_FACTOR = 100;
+    private static final double STANDARD_ERROR = 0.01;
+    private static final double[] quantiles = {0.0001, 0.0200, 0.0300, 0.04000, 0.0500, 0.1000, 0.2000, 0.3000, 0.4000, 0.5000, 0.6000, 0.7000, 0.8000,
+            0.9000, 0.9500, 0.9600, 0.9700, 0.9800, 0.9999};
+
+    private static final Joiner ARRAY_JOINER = Joiner.on(",");
+    private static final MetadataManager METADATA = MetadataManager.createTestMetadataManager();
+
+    @Test
+    public void testNullTDigestGetValueAtQuantile()
+    {
+        functionAssertions.assertFunction("value_at_quantile(CAST(NULL AS tdigest(double)), 0.3)", DOUBLE, null);
+    }
+
+    @Test
+    public void testNullTDigestGetQuantileAtValue()
+    {
+        functionAssertions.assertFunction("quantile_at_value(CAST(NULL AS tdigest(double)), 0.3)", DOUBLE, null);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testGetValueAtQuantileOverOne()
+    {
+        functionAssertions.assertFunction(format("value_at_quantile(CAST(X'%s' AS tdigest(double)), 1.5)",
+                new SqlVarbinary(createTDigest(STANDARD_COMPRESSION_FACTOR).serialize().getBytes()).toString().replaceAll("\\s+", " ")),
+                DOUBLE,
+                null);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testGetValueAtQuantileBelowZero()
+    {
+        functionAssertions.assertFunction(format("value_at_quantile(CAST(X'%s' AS tdigest(double)), -0.2)",
+                new SqlVarbinary(createTDigest(STANDARD_COMPRESSION_FACTOR).serialize().getBytes()).toString().replaceAll("\\s+", " ")),
+                DOUBLE,
+                null);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testInvalidSerializationFormat()
+    {
+        functionAssertions.assertFunction(format("value_at_quantile(CAST(X'%s' AS tdigest(double)), 0.5)",
+                new SqlVarbinary(createTDigest(STANDARD_COMPRESSION_FACTOR).serialize().getBytes()).toString().substring(0, 80).replaceAll("\\s+", " ")),
+                DOUBLE,
+                null);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testEmptySerialization()
+    {
+        functionAssertions.assertFunction(format("value_at_quantile(CAST(X'%s' AS tdigest(double)), 0.5)",
+                new SqlVarbinary(new byte[0])),
+                DOUBLE,
+                null);
+    }
+
+    @Test(invocationCount = 5)
```

Comment:
What's the purpose of setting `invocationCount`? If the test is flaky we shouldn't include it.

Reply:
Since these tests generate different distributions each time, having it run 5 times shows that the digest is robust and behaves as expected, and it's not just getting lucky with a particular distribution.